### PR TITLE
[SPARK-8434][SQL]Add a "pretty" parameter to the "show" method to display long strings

### PR DESCRIPTION
**Pull request by zsxwing**
Sometimes the user may want to show the complete content of cells. Now `sql("set -v").show()` displays:

screen shot 2015-06-18 at 4 34 51 pm

The user needs to use something like `sql("set -v").collect().foreach(r => r.toSeq.mkString("\t"))` to show the complete content.

This PR adds a `pretty` parameter to show. If `pretty` is false, `show` won't truncate strings or align cells right. 

screen shot 2015-06-18 at 4 21 44 pm

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @rxin

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -170,7 +170,7 @@ setMethod("isLocal",
 setMethod("showDF",
           signature(x = "DataFrame"),
           function(x, numRows = 20) {
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: Could you also make this a `pretty = TRUE` here and pass in `pretty` as the argument in `callJMethod` ? The code would look something like 
```
function(x, numRows = 20, pretty = TRUE) {
  s <- callJMethod(x@sdf, "showString", numToInt(numRows), pretty)
}
```

**Comment by andrewor14** [comment]
@davies

**Comment by rxin** [comment]
maybe pretty -> truncate?

**Comment by davies** [comment]
+ 1 for truncate, otherwise LGTM.

**Comment by zsxwing** [comment]
> maybe pretty -> truncate?

I used `truncate` before. But it doesn't imply right alignment. Does it matter?

**Comment by rxin** [comment]
That should be fine, since "pretty" is vague, and the most important factor here is truncation.

**Comment by rxin** [comment]
lgtm

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by rxin** [comment]
Merging. Thanks!

**Pull request closed without merging**